ludwig/models/llm.py

```diff
@@ -95,13 +100,16 @@ def __init__(
         _, self.output_feature_decoder = self.output_features.items()[0]
 
         # ================ Combined loss metric ================
+        ####SXK should we have a class to capture all the loss metrics and output to a dashboard during a run
         self.eval_loss_metric = torchmetrics.MeanMetric()
         self.eval_additional_losses_metrics = torchmetrics.MeanMetric()
 
         clear_data_cache()
 
     def initialize_adapter(self):
         """If an adapter config is provided, we want to wrap the model with a PEFT model for fine-tuning."""
+        ###SXK do we think we can extend the peft module but taylor it more towards ludwig
+        ###what would that look like or will this work out of the box
```

Comment:
This works out of the box already since PEFT is a wrapper on top of transformers, and Ludwig is already compatible with transformers. We could adapt this in the future to a more Ludwig friendly design if we decide we should provide implementations for custom adapters that users might want to design, but I think that's a lot of work and 99.9% users won't want to do that at the moment